docs/proposals/kubemark.md

```diff
@@ -0,0 +1,148 @@
+
+### Elasticsearch
+
+Similarly to Heapster Elasticsearch runs outside the Master machine but generates some traffic on it. Fluentd “daemon” running on Master periodically sends Docker logs it gathered to the Elasticsearch running on one of the Nodes.
+
+In addition Kubelet exposes cadvisor endpoint which is scraped by Heapster, healthz to be read by supervisord, and we have FluentD running as a Pod on each Node that exports
+logs to Elasticsearch (or Google Cloud Logging). Both Heapster and Elasticsearch are running in Pods in the cluster so do not add any load on a Master components by
+themselves. There are other systems that scrape Heapster (cadvisor?) through proxy running on Master, which adds additional load.
+
+This means that our Hollow version of Kubelet will need to have 3 Watch clients, and regularly push Status updates to the Master. In the first version we’ll assume that all started
+Pods will run indefinitely if not explicitly deleted. In the future we can add a model of short-running batch jobs, but in the initial version we’ll assume only serving-like Pods.
+
+We also need to simulate Heapster watching API server, so we’ll need additional process with 3 Watch clients. On top of that we need to create some server to which Fluentd
+will be able to push logs, probably the easiest and the best option is to run real Elasticsearch, which will additionally allow user to inspect Master logs from a test run.
+
+In the first version we won’t simulate scrapers reading Heapster data through proxy on a master machine, or any other data proxies. We don’t know how much load it generates on
+a master machine and it would noticeably complicate the system. If it turns out to be a big source of load we may revisit this decision later.
+
+## Necessary work
+
+There are three more or less independent things that needs to be worked on:
+- HollowNode implementation, creating a library/binary that will be able to listen to Watches and respond in a correct fashion with Status updates. This also involves creation
+of a CloudProvider that can produce such Hollow Nodes.
+- Kubemark setup, including figuring networking model, number of Hollow Nodes that will be allowed to run on a single “machine”, writing setup/run/teardown scripts
+- Creating a Player component that will send requests to the API server putting a load on a cluster. This involves creating a way to specify desired workload. This task is
+very well isolated from the rest, as it is about sending requests to the real API server. Because of that we can discuss requirements separately.
+
+## Concerns
+
+Network performance most likely won't be a problem for the initial version, but when adding additional features it may turn that it becomes one. In such case we may try one of the following:
+- running overlay network like Flannel or OVS instead of using cloud providers routes
+- write simple network multiplexer running on each Node and teach Master to deal with that
```

Comment:
maybe say "simple network multiplexer to multiplex communications from the kubelets/kube proxies on the node"